drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -100,8 +85,10 @@ static inline int stm32_clock_control_on(struct device *dev,
 		LL_APB4_GRP1_EnableClock(pclken->enr);
 		break;
 	default:
+		z_stm32_hsem_release(CFG_HW_RCC_SEMID);
```

Comment:
I won't block for this but a solution idiom of:

```
int rc = 0;

z_stm32_hsem_lock(CFG_HW_RCC_SEMID);
switch (pclken->bus) {
case STM32_CLOCK_BUS_AHB1:
        LL_AHB1_GRP1_EnableClock(pclken->enr);
        break;
...
default:
	rc = -ENOTSUP;
	break;
}

z_stm32_hsem_release(CFG_HW_RCC_SEMID);

return rc;
```

would be easier to understand because there's clearly one unlock for each lock.   It's also more consistent with future style guidance to have one return statement per function.

Reply:
Done